backend/src/main/java/se/kth/iv1201/recruitmentapplicationgroup5/controller/ApiExceptionHandler.java

```diff
@@ -0,0 +1,75 @@
+package se.kth.iv1201.recruitmentapplicationgroup5.controller;
+
+import javax.validation.ConstraintViolationException;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+
+/**
+ * Class for encapsulating non default exception handling in one place.
+ *
+ */
+@ControllerAdvice
+public class ApiExceptionHandler {
+	
+	//Logger here as a private field
+	
+	@ExceptionHandler(value = MethodArgumentNotValidException.class)
+	public ResponseEntity<Object> handleValidationException(
+			MethodArgumentNotValidException e, 
+			WebRequest req)
+	{
+		var status = HttpStatus.BAD_REQUEST;
+		String url = extractUrl(req);
+		String errorMsg = generateErrMsg(e);
+		var body = createErrorBody(status, url, errorMsg);
+
+		return ResponseEntity.status(status).body(body);
+	}
+	
+	@ExceptionHandler(value = ConstraintViolationException.class)
+	public ResponseEntity<Object> handleContraintViolationException(
+			ConstraintViolationException e,
+			WebRequest req)
+	{
+		return handleInternalError(e, req);
+	}
+	
+	@ExceptionHandler(value = Exception.class)
+	public ResponseEntity<Object> handleAllExceptions(Exception e, WebRequest req) {
+		return handleInternalError(e, req);
+	}
+	
+	private ResponseEntity<Object> handleInternalError(Exception e, WebRequest req) {
+		//Log error here instead of System.out
+		System.out.println(e);
+		System.out.println(req);
+						
+		var status = HttpStatus.INTERNAL_SERVER_ERROR;
+		String url = extractUrl(req);
+		String errorMsg = "Something went wrong on our end. Try again.";
+		var body = createErrorBody(status, url, errorMsg);
+		return ResponseEntity.status(status).body(body);
+	}
+	
+	private String extractUrl(WebRequest req) {
+		return ((ServletWebRequest)req).getRequest().getRequestURL().toString();
+	}
+	
+	private String generateErrMsg(MethodArgumentNotValidException e) {
```

Comment:
I think that there are better ways to present the validation errors, for example with a Map with field name as key and error message as value. A string like this would require extra coupling for the View to parse, as it would have to know how the controller writes messages. If a Map was used, the View could just use the field names it already knows to see where the error occurred.

Reply:
Yes I agree. However, these messages should be for developers(of a frontend) and not for display. I think we talked about this with Leif. This would couple messages in the model to the frontend.  It should be an easy fix however, so whatever the group wants can be done.

Reply:
Yes, I didn't mean for the messages to be presented to the end user, those should be for the developers.